### PR TITLE
[Playground] Apache Kafka based Kafka emulator for Playground examples

### LINKED ISSUE #25441: [Task]: Replace confluent-kafka-go Kafka emulator with emulator based on Apache Kafka in Playground
### What needs to happen?

`confluent-kafka-go` built-in emulator is an experimental product with various compatibility issues, e.g. it is not working with current version of Java Kafka client used in Beam (2.4.1)

Replace usage of this emulator with one based on real Apache Kafka server in Playground

### Issue Priority

Priority: 2 (default / most normal work should be filed as P2)

### Issue Components

- [ ] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam examples
- [X] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by TSultanov**
Using [`io.github.embeddedkafka:embedded-kafka`](https://github.com/embeddedkafka/embedded-kafka) for starting a single-process Kafka broker with in-memory storage.

This replaces usage of Kafka server emulator included in `go-confluent-kafka` library, which had shown to have compatibility issues with Java Kafka client used by Beam.

resolves #25441

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: playground/backend/internal/emulators/emulator.go [hunk 1]
```diff
@@ -49,49 +55,51 @@ func PrepareMockClustersAndGetPrepareParams(request *pb.RunCodeRequest) ([]Emula
...
 		case pb.EmulatorType_EMULATOR_TYPE_KAFKA:
-			kafkaMockCluster, err := NewKafkaMockCluster()
+			kafkaMockCluster, err := NewKafkaMockCluster(configuration.KafkaEmulatorExecutablePath)
 			if err != nil {
 				logger.Errorf("failed to run a kafka mock cluster, %v", err)
-				return nil, nil, err
+				return nil, err
 			}
 			mockClusters = append(mockClusters, kafkaMockCluster)
-			datasetDTOs, err := toDatasetDTOs(datasets)
```
Comment (olehborysevych) [inline playground/backend/internal/emulators/emulator.go:63]: i think we can check this before creating mock cluster

File: playground/backend/internal/emulators/kafka.go [hunk 2]
```diff
@@ -31,43 +35,90 @@ import (
...
+	cmd                *exec.Cmd
+	host               string
+	port               string
+	preparerParameters map[string]string
 }
 
-func NewKafkaMockCluster() (*KafkaMockCluster, error) {
-	cluster, err := kafka.NewMockCluster(brokerCount)
+func NewKafkaMockCluster(emulatorExecutablePath string) (*KafkaMockCluster, error) {
+	cmd := exec.Command("java", "-jar", emulatorExecutablePath)
```
Comment (olehborysevych) [inline playground/backend/internal/emulators/kafka.go:55]: Please consider checking cmd.Err before going further

**Comment by olehborysevych** [comment]
LGTM with nits

File: playground/backend/internal/emulators/kafka.go (hunk 2 above)
Comment (TSultanov) [reply playground/backend/internal/emulators/kafka.go:55]: We are building containers with `go1.18`. `os/exec.Cmd` doesn't have `Err` field in that version of Go: https://cs.opensource.google/go/go/+/refs/tags/go1.18.9:src/os/exec/exec.go;l=57

**Comment by TSultanov** [comment]
Run Java PreCommit

**Comment by TSultanov** [comment]
Run Java PreCommit

**Comment by TSultanov** [comment]
Run Java_Kafka_IO_Direct PreCommit

**Comment by TSultanov** [comment]
"Run Java_Examples_Dataflow_Java11 PreCommit"

**Comment by Abacn** [comment]
Thanks, will looking into it today

File: playground/backend/cmd/server/controller.go [hunk 3]
```diff
@@ -15,6 +15,7 @@
 package main
 
 import (
+	"beam.apache.org/playground/backend/internal/emulators"
```
Comment (Abacn) [inline playground/backend/cmd/server/controller.go:18]: minor: unnecessary move of import package. Wondering that doesn't playground code have go fmt check?

**Review by Abacn — COMMENTED** [review]
Thanks, this LGTM, just one minor question

**Pull request merged**